tests/testthat/test-thumbnail.R

```diff
@@ -0,0 +1,120 @@
+
+for (version in names(mock_dirs)) {
+  with_mock_dir(mock_dirs[[version]], {
+    with_mock_api({
+      test_that(glue::glue("get_thumbnail() gets the thumbnail ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "01234567")
+        
+        # User-specified path
+        user_path <- tempfile("thumbnail_", fileext = ".jpg")
+        received_path <- get_thumbnail(item, user_path)
+        received <- readBin(received_path, "raw", n = 8)
+        expected <- as.raw(c(0x4e, 0x41))
+        expect_equal(user_path, received_path)
+        expect_equal(received, expected)
+        
+        # Automatic path
+        received_path <- get_thumbnail(item)
+        received <- readBin(received_path, "raw", n = 8)
+        expected <- as.raw(c(0x4e, 0x41))
+        expect_equal(substring(received_path, nchar(received_path) - 4), ".jpeg")
+        expect_equal(received, expected)
+        })
+    
+      test_that(glue::glue("get_thumbnail() returns NA_character_ for 204 status codes ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "12345678")
+        
+        # User-specified path
+        user_path <- tempfile("thumbnail_", fileext = ".jpg")
+        received_path <- get_thumbnail(item, user_path)
+        expect_equal(received_path, NA_character_)
+        
+        # Automatic path
+        received_path <- get_thumbnail(item)
+        expect_equal(received_path, NA_character_)
+      })
+    
+      test_that(glue::glue("get_thumbnail() errors with 404 status codes ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "23456789")
+        expect_error(get_thumbnail(item), "request failed with Client error: \\(404\\) Not Found")
+      })
+    
+      test_that(glue::glue("has_thumbnail() returns TRUE when the item has a thumbnail ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "01234567")
+        expect_true(has_thumbnail(item))
+      })
+    
+      test_that(glue::glue("has_thumbnail() returns FALSE when the status code is 204 ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "12345678")
+        expect_false(has_thumbnail(item))
+      })
+    
+      test_that(glue::glue("has_thumbnail() errors with 404 status codes ({version})"), {
+        client <- connect(server = "https://connect.example", api_key = "fake")
+        item <- content_item(client, "23456789")
+        expect_error(has_thumbnail(item), "request failed with Client error: \\(404\\) Not Found")
+      })
+    
+      test_that(glue::glue("set_thumbnail() returns returns a content item when successful ({version})"), {
```

Comment:
```suggestion
      test_that(glue::glue("set_thumbnail() returns returns the content item when successful ({version})"), {
```

Is this more correct? Or maybe I don't understand what's being tested?

Reply:
Yes. Accepted the change locally.